cloud/passportauth/src/auth/PassportAuth.ts

```diff
@@ -0,0 +1,54 @@
+import * as express from 'express';
+import * as passport from 'passport';
+import BaseUser from '../user/BaseUser';
+import UserSecService from '../user/UserSec';
+
+interface Auth {
+  protect(req: express.Request, res: express.Response, next: () => void): void; // protect/authenticate?
```

Comment:
Actual method may be:
`protect(role:string) => (req: express.Request, res: express.Response, next: () => void) => void;`